eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Ledger.hs

```diff
@@ -211,15 +222,21 @@ instance
   initialRules = []
   transitionRules = [ledgerTransition @ConwayLEDGER]
 
-  renderAssertionViolation AssertionViolation {avSTS, avMsg, avCtx, avState} =
-    "AssertionViolation ("
-      <> avSTS
-      <> "): "
-      <> avMsg
-      <> "\n"
-      <> show avCtx
-      <> "\n"
-      <> show avState
+  renderAssertionViolation AssertionViolation {..} =
+    unlines
+      [ "Assertion violation:"
+      , show avSTS
+      , show avMsg
+      , renderFailure avMsg
+      ]
+    where
+      TRC (_, LedgerState {lsUTxOState = utxoSt, lsCertState = certState}, _) = avCtx
+      renderFailure "Deposit pot must equal obligation (LEDGER)" =
+        unlines
+          [ "utxosDeposited: " <> show (utxoSt ^. utxosDepositedL)
+          , "totalObligation: " <> show (totalObligation certState (utxoSt ^. utxosGovStateL))
+          ]
+      renderFailure _ = ""
```

Comment:
I can't see any other ways for pretty printing failures when they happen so that I can also access `TRC`. What should I put there?

Reply:
I meant rendering it in the LEDGER rule, rather than here:
https://github.com/input-output-hk/cardano-ledger/blob/d883cf700488ed1c3ecefd403c9863a0e3943587/eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Ledger.hs#L277-L299

Reply:
So, unifying the same way rendering happens in Shelley and in Conway.
I don't understand why we need to have different renderings for the same assertion in two different eras

Reply:
I did it because I also want information about the actual values of `utxosDeposited` and `totalObligation`. This information helped me to debug what was going wrong with submitting transactions.

Reply:
Oh ok, I see that's what they're doing there already, I'll take a look at that function.